test/base/src/androidTest/kotlin/io/realm/shared/RealmListTests.kt

```diff
@@ -368,7 +369,7 @@ internal interface ListApiTester {
     fun errorCatcher(block: () -> Unit) {
         try {
             block()
-        } catch (e: AssertionError) {
+        } catch (e: RealmCoreException) {
```

Comment:
I guess this is infrastructural code, so it should in fact catch `AssertionError` ... shouldn't it?